client/state/reader/follows/reducer.js

```diff
@@ -20,13 +20,23 @@ import {
 import { prepareComparableUrl } from './utils';
 import { createReducer } from 'state/utils';
 
-function updatePostSubscription( state, payload, shouldSkipUpdate, newProps ) {
+/**
+ * Update a post subscription
+ *
+ * @param {object} state The current state from the items reducer
+ * @param {object} payload the payload from the subscribe / update / unsubscribe action
+ * @param {function} isEqualToCurrentState A function that determines if the payload will actually change the current state.
+ * Returns true if the state would not change.
+ * @param {object} newProps The new props to set on the delivery_methods.email object if the state will change
+ * @returns {object} the next state
+ */
+function updatePostSubscription( state, payload, isEqualToCurrentState, newProps ) {
 	const follow = find( state, { blog_ID: +payload.blogId } );
 	if ( ! follow ) {
 		return state;
 	}
 
-	if ( shouldSkipUpdate( follow, payload ) ) {
+	if ( isEqualToCurrentState( follow, payload ) ) {
```

Comment:
and i say that because, while this check is pretty cheap, changing the instance means changing the state all the way up the state tree, firing a change event on the redux store, re-running all of the selectors and state to props mappers across the entire app, when nothing changed.